File: Python/BubbleSort.py
# Python3 Optimized implementation of Bubble sort


def bubble_sort(arr):
    n = len(arr)

    # Traverse through all array elements
    for i in range(n):
        swapped = False

        # Last i elements are already in place
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break


# Driver code to test above
if __name__ == "__main__":
    arr = [64, 34, 25, 12, 22, 11, 90]

    print("Array:", arr)

    bubble_sort(arr)
    print("Sorted array:", arr)

File: Cpp/BubbleSort.cpp
#include <bits/stdc++.h>
#include "sorting_algorithms.h"
 
using namespace std; 

// A function to implement bubble sort 
void bubbleSort(int arr[], int n) 
{ 
	int i, j, is_swap = 0; 
	for (i = 0; i < n-1; i++)	 
	{
		is_swap=0;
		// Last i elements are already in place 
		for (j = 0; j < n-i-1; j++) 
			if (arr[j] > arr[j+1]) 
			{	
				swap(arr[j], arr[j+1]); 
				is_swap=1;
			}
		if(is_swap==0)
			break;
	}
} 


